src/AIS.cpp

```diff
@@ -234,3 +235,88 @@ ais_base::Position AIS::applyPositionCorrection(ais_base::Position const& sensor
 
     return vessel_pos;
 }
+
+/**
+ * @brief General template for handling values that might be unknown, returning a default
+ * value.
+ */
+template <typename SourceType>
+SourceType safe_value(SourceType value, SourceType default_value)
+{
+    return (value == base::unknown<SourceType>()) ? default_value : value;
```

Comment:
It is wrong to check for NaN equality. I.e. two NaN values can be different

```suggestion
    return (base::isUnknown(value)) ? default_value : value;
```